### ISSUE TITLE
Grayscale emulation possible?

### ISSUE THREAD

**Issue by jose1711** [LTVA1/flipper-zero-video-player#14]
Would video player be able to emulate grayscale as in https://www.reddit.com/r/flipperzero/comments/10uocof/update_on_grayscale_emulation_more_details_in/ or https://www.reddit.com/r/flipperzero/comments/10qztme/my_next_project_is_grayscale_emulation_for_the/?

**Comment by LTVA1** [comment]
I am not sure, because it requires precise timing while video player spends a lot of time reading data from SD card and drawing it. I was actually experimenting with alternating checkerboard pattern dithering, running it off hardware interrupt and with direct draw, but it still flickered no matter how I tried. I would ask the author of greyscale camera app to actually shoot a video showcasing the greyscale, since with appropriate shutter speed ypu can capture good looking photo of it, but in reality it most probably flickers.

By the way, the speed at which LCD pixels change their color depends on the temperature too.

**Comment by GooseDev72** [comment]
@LTVA1 I made it, uses animation (something like checkerboard), moving only neighbor pixels, so I got grayscale emulation, can you add it to the readme if I drop it on my github? Works with latest vidplayer and firmware

**Comment by LTVA1** [comment]
Can you please upload the fap here? I may test it tomorrow

**Comment by GooseDev72** [comment]
@LTVA1 
No, that is the script that generates .bnd with grayscale emulation (the vid player stays intact)

**Comment by GooseDev72** [comment]
Python .mp4 to .bnd converter

**Comment by LTVA1** [comment]
Ah then upload it on github, I can test it on some videos

**Comment by GooseDev72** [comment]
@LTVA1 https://github.com/GooseDev72/my-hacking-goodies/blob/main/scripts/flipper/fzvidmkr.py

**Comment by GooseDev72** [comment]
Flickers a little in reality, but that is the best I could do. I could try working on the application and reimplementing it in u8g2 calls

**Comment by LTVA1** [comment]
well it kinda works, but you are welcome to reimplement the drawing routine properly on timer interrupts and without `canvas_draw_xbm` which eats around 80% of the CPU time because it also can do some decompression. I wonder if you can pinpoint the exact screen refresh frequency and match timer frequency to artificially dither the video frames, but that would require some changes in the file format (i.e. pictures would be at least 2bpp instead of 1bpp)
